translate/pootle
# Allauth-based revamped auth workflow

This PR introduces the allauth-based authentication backend, and implements the new workflow and UI for it.

Fixes #3343, #3515, #3517, #3518, #3728, #3737, #3774.

**IMPORTANT NOTES**
* It's **highly suggested** to migrate existing users to Allauth's own way to manage email addresses.
* The commit messages have issue references with stripped `#`s to avoid spam. They'll be brought back to normal immediately before merging happens.
* The PR doesn't cover:
  * Email management (add/change/remove addresses)
  * Explicit account linking/unlinking features

## Workflow testing

Before testing, ensure you are up-to-date with npm dependencies (`npm install` within *pootle/static/js*) and make sure to build JS modules (`manage.py webpack --dev`).

An outline of possible tests of the workflow follows. Added to this, triggering form validation errors anywhere in the process can help too.

1. Sign In
  * Confirmed user (signed up, verified email address)
     * signs user in
        * redirects to
          * previous URL in case `next` was specified
          * `settings.LOGIN_URL` otherwise
  * Unconfirmed user (signed up, didn't verify email address yet)
    * re-sends activation email
    * re-check this after a cooling down period of a few minutes; prevents impatient users from being bombarded with emails
  * Unconfirmed user (existed prior to allauth, didn't verify email address in allauth) — *this should ideally be avoided by migrating users' data*
    * sends activation email
    * click email link to verify email and confirm user account
  * Inactive user (admin deactivated user, `is_active = False`)
    * trying to sign in will display the inactive screen
2. Sign Up
  * Provide new user data
    * sends activation email
  * Click email link to verify email and confirm user account
    * signs user automatically in
    * redirects to `settings.LOGIN_REDIRECT_URL`
3. Password Reset
  * Request password reset
    * sends reset link in an email
  * Click reset link in the email
    * reset password
      * resets password and signs user in, redirecting to `/` (unfortunately cannot redirect to `settings.LOGIN_REDIRECT_URL` due to allauth's limitation)
    * sign out
    * sign in with new password
  * Open invalid/outdated/already-reset password reset link
    * displays message of invalid reset link

If you feel brave you can also test some social authentication provider (just enable a provider by adding it to `INSTALLED_APPS`; it'll show up), but note that that area hasn't been fully addressed yet so YMMV.
